### PR TITLE
OTC-887: Display proper information when validating email

### PR DESCRIPTION
**Pull request by olewandowski1**
**[CORE PR MERGE REQUIRED](https://github.com/openimis/openimis-fe-core_js/pull/112)**

[OTC-887](https://openimis.atlassian.net/browse/OTC-887)

Changes:
- Possibility to display information that value format is invalid implemented.
- If the email format is invalid, user will get an information about that and it is not possible to save.
- New translation key has been added to the Lokalise.

[OTC-887]: https://openimis.atlassian.net/browse/OTC-887?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ export const CLAIM_ADMIN_USER_TYPE = "CLAIM_ADMIN";
 export const CLAIM_ADMIN_IS_SYSTEM = 256;
 export const MODULE_NAME = "user";
 
+export const EMAIL_REGEX_PATTERN = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (delcroip) [inline src/constants.js:25]: Can you fetch the regex from BE (exposed configuration ? )

Comment (delcroip) [reply src/constants.js:25]: https://github.com/openimis/openimis-be-core_py/pull/164/files

Comment (olewandowski1) [reply src/constants.js:25]: Do you mean that we should expose an endpoint and fetch if the format is valid or not? @delcroip

Comment (delcroip) [reply src/constants.js:25]: no just the regex for email or at least copy it from be

Comment (delcroip) [inline src/constants.js:25]:
```suggestion
export const EMAIL_REGEX_PATTERN = /^([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\[[\t -Z^-~]*])$/;
```

Comment (olewandowski1) [reply src/constants.js:25]: We have conducted a more accurate research and found a better regex than the previous one. Let us know if it is acceptable for you. If so, we'll update both BE and FE to use it.
**`export const EMAIL_REGEX_PATTERN =
  /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;`**

Comment (olewandowski1) [reply src/constants.js]: Thanks, I've changed this Regex. @delcroip

**Review by delcroip — APPROVED** [review]

**Comment by delcroip** [comment]
please fix the conflict

**Comment by olewandowski1** [comment]
@delcroip Sure, conflict solved.

**Pull request merged**